legacy/PimStructure.js

```diff
@@ -783,6 +854,92 @@ class PimStructure {
     return childMap;
   }
 
+  async updateExportRecordsWithVariantValueIds(valuesList, exportRecords) {
+    let vvIdNameMap = new Map();
+    for (let variantValue of valuesList) {
+      vvIdNameMap.set(variantValue.Name, variantValue.Id);
+    }
+    for (let record of exportRecords) {
+      const recordName = record.get('Record_ID');
+      if (recordName) {
+        record.set('Id', vvIdNameMap.get(recordName));
+      }
+    }
+  }
+
+  async getFinalizedDaList(
```

Comment:
I am not sure what to do here or what advice to put on this PR, because you are dealing with code we need to redo...but.
In general this function does to much. One way you can see that it's preforming to many logical functions is the big-ass `if` statement at the start. `if (isInherited...` Typically you don't want to split your functions like this and if you have something like this you would need to break this into other functions. 
```
async getFinalizedDaList() {
  if (isInherited) {
    this.processDaListInherited(data);
  } else {
    this.processDaList(data);
  }
```
or something like this but just to break up the logic so you can reuse stuff if needed.

And this function has way to many parameters which is difficult to know what is going on. You would want to pass in an object with all these as attributes. https://stackoverflow.com/questions/2432443/best-practice-for-passing-many-arguments-to-method

With all this said I am not sure it's worth updating the code you wrote in these files. Already this file is huge and needs to be refactored/redone so it might not be worth putting in this extra work.

Up to you.

Reply:
Ok, did a partial refactor to break up the function. Will handle the rest when I refactor the whole of export